Misuse of `OwnedFd`?

When trying to compile against commit 8ab3463467c2121e1c934232fd59398dcbea1600:

```
error[E0382]: use of moved value: `fd`
   --> smithay/src/backend/allocator/gbm.rs:138:17
    |
132 |         let fd = self.fd()?;
    |             -- move occurs because `fd` has type `OwnedFd`, which does not implement the `Copy` trait
...
138 |                 fd,
    |                 ^^ value moved here, in previous iteration of loop
```

rustc 1.68.0-nightly (ee0412d1e 2023-01-07)

--- a/src/backend/allocator/gbm.rs
+++ b/src/backend/allocator/gbm.rs
@@ -124,15 +124,13 @@ impl<T> AsDmabuf for GbmBuffer<T> {
             // check all planes have the same handle. We can't use
             // drmPrimeHandleToFD because that messes up handle ref'counting in
             // the user-space driver.
-            return Err(GbmConvertError::UnsupportedBuffer); //TODO
+            return Err(GbmConvertError::UnsupportedBuffer);
         }
-
-        // Make sure to only call fd once as each call will create
-        // a new file descriptor which has to be closed
-        let fd = self.fd()?;
 
         let mut builder = Dmabuf::builder_from_buffer(self, DmabufFlags::empty());
         for idx in 0..planes {
+            let fd = self.fd()?;
+
             builder.add_plane(
                 // SAFETY: `gbm_bo_get_fd` returns a new fd owned by the caller.
                 fd,

